primitives/runtime/src/lib.rs

```diff
@@ -97,6 +97,9 @@ pub use sp_arithmetic::{
 
 pub use either::Either;
 
+/// The maximum depth for a nested pallet error enum.
+pub const MAX_PALLET_ERROR_ENCODED_SIZE: usize = 4;
```

Comment:
Maybe we can have a more wider definition, because this is in primitive and not part of FRAME.
Maybe something like: "The number of byte of a ModuleError, in FRAME it is the maximum depth for a nested pallet error enum."
Anyhow it is also fine as it is.